Services/Contact/classes/class.ilMailMemberSearchGUI.php

```diff
@@ -1,66 +1,47 @@
-<?php
+<?php declare(strict_types=1);
 /* Copyright (c) 1998-2015 ILIAS open source, Extended GPL, see docs/LICENSE */
 
+use Psr\Http\Message\RequestInterface;
+
 /**
  * Class ilMailMemberSearchGUI
  * @author Nadia Matuschek <nmatuschek@databay.de>
  *
 **/
 class ilMailMemberSearchGUI
 {
-    /**
-     * @var mixed
-     */
-    protected $mail_roles;
-
+    private RequestInterface $httpRequest;
+    protected mixed $mail_roles;
     /**
      * @var ilObjGroupGUI|ilObjCourseGUI
      */
-    protected $gui;
-
-    /**
-     * @var ilAbstractMailMemberRoles
-     */
-    protected $objMailMemberRoles;
+    protected mixed $gui;
+    protected ilAbstractMailMemberRoles $objMailMemberRoles;
     /**
      * @var null object ilCourseParticipants || ilGroupParticipants
      */
-    protected $objParticipants = null;
-
-    /**
-     * @var ilCtrl
-     */
-    protected $ctrl;
-
-    /**
-     * @var ilTemplate
-     */
-    protected $tpl;
-
-    /**
-     * @var ilLanguage
-     */
-    protected $lng;
-    
-    /**
-     * @var ilAccessHandler
-     */
-    protected $access;
+    protected mixed $objParticipants;
```

Comment:
The `mixed` type is not supported in PHP 7.4, which we'll have to support as well in ILIAS 8.